src/FSharpLint.Client/LSPFSharpLintServiceTypes.fs

```diff
@@ -0,0 +1,71 @@
+module FSharpLint.Client.LSPFSharpLintServiceTypes
+
+open System
+open System.Diagnostics
+open System.IO
+open StreamJsonRpc
+
+type FSharpLintResponseCode =
+    | ErrToolNotFound = -5
+    | ErrFileNotFound = -4
+    | ErrFilePathIsNotAbsolute = -3
+    | ErrCancellationWasRequested = -2
+    | ErrDaemonCreationFailed = -1
+    | OkCurrentDaemonVersion = 0
+
+type FSharpLintVersion = FSharpLintVersion of string
+type FSharpLintExecutableFile = FSharpLintExecutableFile of string
+type Folder = private Folder of string 
+with 
+    static member from (filePath: string) =
+        if File.Exists(filePath) then 
+            // Path.GetFullPath to resolve path like /foo/bar/../baz
+            let folder = ((filePath |> Path.GetFullPath |> FileInfo).Directory).FullName
+            if DirectoryInfo(folder).Exists then
```

Comment:
@MrLuje there's an unnecessary allocation here, the `Directory` property of line 23 is already a DirectoryInfo instance, so you can just use its `Exists` property directly instead of creating another instance.

Reply:
whoops, you're right